src/components/hooks/use-handle-lazy-load.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { ReactNode, useEffect, useState } from 'react'
+
+export const useHandleLazyLoad = (lazyTemplates: ReactNode, targetId: string, options: object) => {
+    const [is_visible, setIsVisible] = useState(false)
+    const [is_large_screen, setLargeScreen] = useState(false)
+
+    const handleIntersect = (entries, observer) => {
+        const [entry] = entries
+        if (typeof entry.isVisible === 'undefined') {
+            // The browser doesn't support Intersection Observer v2, falling back to v1 behavior.
+            setIsVisible(entry.isIntersecting)
+            observer.disconnect()
+            return
+        }
+        if (entry.isIntersecting) {
+            setIsVisible(entry.isIntersecting)
+            observer.disconnect()
+        }
+    }
+    useEffect(() => {
+        createObserver()
+        if (window.innerHeight >= 950) {
+            setLargeScreen(true)
+        }
+    }, [])
+
+    const createObserver = () => {
+        const target = document.querySelector(targetId)
+        const observer = new IntersectionObserver(handleIntersect, options)
+        observer.observe(target)
+    }
+
+    return is_visible || is_large_screen ? (
+        <>{lazyTemplates}</>
+    ) : (
+        <>
+            <span>loading</span>
```

Comment:
We need to do something with this loading text. It looks bad . Maybe need some loader or something